extend-string.js

```diff
@@ -0,0 +1,114 @@
+/* Checks if a string contains vowels */
+String.prototype.hasVowels = function () {
+  return (/[aeiou]/g).test(this); //use .test to match regEx patter against string to return boolean
+};
+
+/*Converts entire string to uppercase letter*/
+String.prototype.toUpper = function () {
+  let capitalizedString = ""; //empty string variable to store result
+  for(let index = 0; index < this.length; index++) { //loop through the string to get each letter
+    let characterCode = this.charCodeAt(index); //get the character code of each character
+    if(characterCode >= 97 && characterCode <= 122) { //check if character code is in lowercase using ascii code
+      characterCode -= 32; //subtract 32 from character code to get uppercase equivalent character code
+    }
+    capitalizedString += String.fromCharCode(characterCode); //get letter from character code
+  }
+  return capitalizedString;
+};
+
+/*Converts entire string to lowercase letter*/
+String.prototype.toLower = function () {
+  let lowercaseString = ""; //empty string variable to store result
+  for(let index = 0; index < this.length; index++) { //loop through the string to get each letter
+    let characterCode = this.charCodeAt(index); //get the character code of each character
+    if(characterCode >= 65 && characterCode <= 90) { //check if character code is in uppercase using ascii code
+      characterCode += 32; //add 32 from character code to get lowercase equivalent character code
+    }
+    lowercaseString += String.fromCharCode(characterCode); //get character equivalent from character code
+  }
+  return lowercaseString;
+};
+
+/*Changes the first letter of a string to uppercase*/
+String.prototype.ucFirst = function () {
+  if(/^[a-z]/.test(this)) { //check if string starts with a small letter
+    let stringArray = this.split(""); //split the string to an array
+    stringArray[0] = stringArray[0].toUpper(); //convert first letter to uppercase
+    return stringArray.join(""); //Join the array with no special character to get string
+  }
+  return "First letter is already uppercase"; //return an error if string doesn't start with lowercase letter
+};
+
+/*Checks if the string is a question*/
+String.prototype.isQuestion = function () {
+  return (/.+\?$/g).test(this); //pattern returns boolean if string ends with a ?.
+};
+
+/*Gets all the words in a string*/
+String.prototype.words = function () {
+  return this.match(/[a-zA-z0-9]+/g); //return an array of words that match regEx pattern.
+};
+
+/*Gets the amount of word in a string*/
+String.prototype.wordCount = function () {
+  return this.words().length; //return the length of array of words gotten from a string
+};
+
+/*Gets currency representation of a string*/
+String.prototype.toCurrency = function () {
+  return this.replace(/\d(?=(\d{3})+\.)/g, '$&,'); //replace patter result with pattern result + , (result,)
```

Comment:
The first and second regex pattern passed as parameters are really looking complex. Please extract them into properly named variables and use the variables here instead.